driver-core/src/main/com/mongodb/client/model/CountOptions.java

```diff
@@ -31,9 +31,9 @@
 public class CountOptions {
     private Bson hint;
     private String hintString;
-    private long limit;
+    private int limit;
```

Comment:
It's weird that these would be of type long in CountOptions yet of type int in FindOptions (where they have to be int because of the wire protocol: http://docs.mongodb.org/meta-driver/latest/legacy/mongodb-wire-protocol/#op-query).

@craiggwilson was this intentional?